demo/site/src/app/[[...path]]/page.tsx

```diff
@@ -22,15 +21,20 @@ export default async function Page({ params }: { params: { path: string[] } }) {
     if (draftMode().isEnabled) {
         previewData = { includeInvisible: false };
     }
-    const client = createGraphQLClient(previewData);
+    const graphqlFetch = createGraphlFetchWithPreviewHeaders(fetch, previewData);
+
     const locale = /*context.locale ??*/ defaultLanguage;
     const scope = { domain, language: locale };
 
     //fetch documentType
-    const data = await client.request<GQLDocumentTypeQuery, GQLDocumentTypeQueryVariables>(documentTypeQuery, {
-        path: `/${(params.path ?? []).join("/")}`,
-        scope,
-    });
+    const data = await graphqlFetch<GQLDocumentTypeQuery, GQLDocumentTypeQueryVariables>(
+        documentTypeQuery,
+        {
+            path: `/${(params.path ?? []).join("/")}`,
+            scope,
+        },
+        { next: { revalidate: 3 } },
```

Comment:
What is the default behavior for revalidation? 

In the [Next docs](https://nextjs.org/docs/app/building-your-application/caching#duration-1), it sounds to me like the Data Cache duration is infinite if revalidate isn't set. That would be bad if setting revalidate is forgotten. So maybe we should set a sensible default?

Reply:
yes, but that's how next works. Do you really think we should change the default behavior?

Reply:
I created a new decorator function:

`createFetchWithDefaults(fetch, { next: { revalidate: 15 * 60 } }),`